lab-andrew/lib/client-pool.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+
+function random() {
+  return Math.floor(Math.random() * 5);
+}
+var nickNames = ['Troll', 'Mr. Smith', '42', 'Curry', 'Matt but better'];
+
+function registerClient(socket, clientPool) {
+  socket.wack = {};
+  socket.wack.id = nickNames[random()];
```

Comment:
The assignment specifies that a random id be used, while this is "technically" correct, it won't work if there are more than 5 users since some ids may overlap.  This could also potentially happen with as few as 2 users.  I suggest using the node-uuid library to generate a random alphanumeric string and use that as a client's id.